app/src/test/java/com/example/habitsmasher/test/DatePickerDialogFragmentTest.java

```diff
@@ -0,0 +1,130 @@
+package com.example.habitsmasher.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import com.example.habitsmasher.DatePickerDialogFragment;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Calendar;
+import java.util.Date;
+
+public class DatePickerDialogFragmentTest {
+    //For testing parseStringToDate
+    private Date _dateStringToDate;
+    private String _stringStringToDate;
+    private Calendar _calendarForStringToDate = Calendar.getInstance();
+
+    //For testing parseDateToString for current date
+    private Date _dateDateToStringToday;
+    private String _stringDateToStringToday;
+    private Calendar _calendarForDateToStringToday = Calendar.getInstance();
+
+    //For testing parseDateToString for past date
+    private Date _dateDateToStringPast;
+    private String _stringDateToStringPast;
+    private Calendar _calendarForDateToStringPast = Calendar.getInstance();
+
+    //For testing parseDateToString for future date
+    private Date _dateDateToStringFuture;
+    private String _stringDateToStringFuture;
+    private Calendar _calendarForDateToStringFuture = Calendar.getInstance();
```

Comment:
You don't actually need to initialize all of these separately since you're also defining them later , it might be easier to just have like
```
private Date _testDate;
private String _testDateString ;
private Calendar  _testCalender = Calendar.getInstance(); 
```
especially since you're using the same  calendar each time. Then just set the variables to the specific dates within the tests themselves. It may improve readability. Might also just be my personal preference though so I'll leave it up to you!